bytecomp/bytesections.ml

```diff
@@ -15,31 +15,65 @@
 
 (* Handling of sections in bytecode executable files *)
 
-(* List of all sections, in reverse order *)
-
-let section_table = ref ([] : (string * int) list)
+module Name = struct
+  type t = string
+  let of_string name =
+    if String.length name <> 4 then
+      invalid_arg "Bytesections.Name.of_string: must be of size 4";
+    name
+  let code = of_string "CODE"
+  let dlpt = of_string "DLPT"
+  let dlls = of_string "DLLS"
+  let data = of_string "DATA"
+  let prim = of_string "PRIM"
+  let symb = of_string "SYMB"
+  let dbug = of_string "DBUG"
+  let crcs = of_string "CRCS"
+  let rntm = of_string "RNTM"
+end
```

Comment:
My first reaction would have been to use a sum type
```
type name = CODE | DLPT | DLLS | ...
```
which could be exposed directly, together with string conversions with would be used only internally. Also, this makes it nicer to pattern match on the name.

Reply:
I though about it but the current API allows one to store/read arbitrary section which I though could be useful.
One could use `type name = CODE | DLPT | DLLS | ... | Other of string` but then parttern marching is no longer nice because you might want to match `| CODE | Other "CODE"`. Also, if you don't want to allow `Other of string`,  you'll have to fail if you read an unrecognize section name.

Reply:
I don't understand why you would need to pattern match on `CODE | Other "CODE"`: the `of_string` function will enforce that it will never return `Other "CODE"`, right? As far as I can see, having a `Other` type would be fine and permit using a sum type both for constructing and pattern matching on section names.

I don't want to insist too much on this, but I find the current code that uses module values for the standard section names but that still pattern matches on the naked string literals a bit ugly.

Reply:
I've pushed a commit to use variants